src/main/java/org/benetech/servicenet/repository/ActivityRepository.java

```diff
@@ -216,11 +230,23 @@ private Predicate addDateFilter(FiltersActivityDTO filtersActivityDTO, Predicate
         Join<ActivityInfo, OrganizationMatch> matchJoin = root.join(ORGANIZATION_MATCHES, JoinType.LEFT);
         predicate = cb.and(predicate, cb.equal(matchJoin.get(HIDDEN), filtersActivityDTO.getHiddenFilter()));
 
-        if (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList())) {
+        boolean hasPartnerFilters = (CollectionUtils.isNotEmpty(filtersActivityDTO.getPartnerFilterList()));
+        if (partnerRecords || hasPartnerFilters) {
             Join<OrganizationMatch, Organization> matchOrgJoin = matchJoin.join(PARTNER_VERSION, JoinType.LEFT);
-            Join<Organization, SystemAccount> accountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
-
-            predicate = cb.and(predicate, accountJoin.get(ID).in(filtersActivityDTO.getPartnerFilterList()));
+            Join<Organization, SystemAccount> matchAccountJoin = matchOrgJoin.join(ACCOUNT, JoinType.LEFT);
+            if (partnerRecords) {
+                predicate = cb.and(predicate, cb.or(cb.equal(root.get(ACCOUNT_ID), ownerId),
```

Comment:
I think you can remove `cb.equal(root.get(ACCOUNT_ID), ownerId)`, because organization cannot have a match on organization with the same account, so `cb.notEqual(matchAccountJoin.get(ID), ownerId)` will be always true for current account organizations